FEXCore/Source/Interface/Core/Dispatcher/Dispatcher.cpp

```diff
@@ -590,6 +617,427 @@ void Dispatcher::ExecuteJITCallback(FEXCore::Core::CpuStateFrame* Frame, uint64_
+
+  constexpr static auto VABI1 = ARMEmitter::VReg::v0;
+  constexpr static auto VABI2 = ARMEmitter::VReg::v1;
+
+  auto FillF80Result = [&]() {
+    if (VTMP1 != VABI1) {
+      mov(VTMP1.Q(), VABI1.Q());
+    }
+    FillForABICall(CTX->HostFeatures.SupportsPreserveAllABI, true);
+  };
+
+  auto FillF64Result = [&]() {
+    if (VTMP1 != VABI1) {
+      fmov(VTMP1.D(), VABI1.D());
+    }
+    FillForABICall(CTX->HostFeatures.SupportsPreserveAllABI, true);
+  };
+
+  auto FillF32Result = [&]() {
+    if (VTMP1 != VABI1) {
+      fmov(VTMP1.S(), VABI1.S());
+    }
+    FillForABICall(CTX->HostFeatures.SupportsPreserveAllABI, true);
+  };
+
+  auto FillI64Result = [&]() {
+    if (TMP1 != ABI1) {
+      mov(TMP1, ARMEmitter::XReg::x0);
+    }
+    FillForABICall(CTX->HostFeatures.SupportsPreserveAllABI, true);
+  };
+
+  auto FillI32Result = [&]() {
+    if (TMP1 != ABI1) {
+      mov(TMP1.W(), ARMEmitter::WReg::w0);
+    }
+    FillForABICall(CTX->HostFeatures.SupportsPreserveAllABI, true);
+  };
+
+  auto FillI16Result = [&]() {
+    if (TMP1 != ABI1) {
+      mov(TMP1, ARMEmitter::XReg::x0);
+    }
+    FillForABICall(CTX->HostFeatures.SupportsPreserveAllABI, true);
+  };
+
+  switch (ABI) {
+  case FABI_F80_I16_F32_PTR: {
+    // Linux Reg/Win32 Reg:
+    // tmp4 (x4/x13): FallbackHandler
+    // x30: return
+    // vtmp1 (v0/v16): source
+    SpillForABICall(CTX->HostFeatures.SupportsPreserveAllABI, TMP3, true);
+
+    if (VTMP1 != VABI1) {
+      fmov(VABI1.S(), VTMP1.S());
+    }
+    ldrh(ARMEmitter::WReg::w0, STATE, offsetof(FEXCore::Core::CPUState, FCW));
+    mov(ARMEmitter::XReg::x1, STATE);
+    if (!CTX->Config.DisableVixlIndirectCalls) [[unlikely]] {
+      GenerateIndirectRuntimeCall<FEXCore::VectorRegType, uint16_t, float, uint64_t>(FallbackPointerReg);
+    } else {
+      blr(FallbackPointerReg);
+    }
+
+    FillF80Result();
+  } break;
+  case FABI_F80_I16_F64_PTR: {
+    // Linux Reg/Win32 Reg:
+    // tmp4 (x4/x13): FallbackHandler
+    // x30: return
+    // vtmp1 (v0/v16): source
+    SpillForABICall(CTX->HostFeatures.SupportsPreserveAllABI, TMP3, true);
+
+    if (VTMP1 != VABI1) {
+      fmov(VABI1.D(), VTMP1.D());
+    }
+    ldrh(ARMEmitter::WReg::w0, STATE, offsetof(FEXCore::Core::CPUState, FCW));
+    mov(ARMEmitter::XReg::x1, STATE);
+    if (!CTX->Config.DisableVixlIndirectCalls) [[unlikely]] {
+      GenerateIndirectRuntimeCall<FEXCore::VectorRegType, uint16_t, double, uint64_t>(FallbackPointerReg);
+    } else {
+      blr(FallbackPointerReg);
+    }
+
+    FillF80Result();
+  } break;
+  case FABI_F80_I16_I16_PTR:
+  case FABI_F80_I16_I32_PTR: {
+    // Linux Reg/Win32 Reg:
+    // tmp4 (x4/x13): FallbackHandler
+    // x30: return
+    // tmp2 (x1/x11): source
+    SpillForABICall(CTX->HostFeatures.SupportsPreserveAllABI, TMP3, true);
+
+    if (TMP2 != ABI2) {
+      if (ABI == FABI_F80_I16_I16_PTR) {
+        sxth(ARMEmitter::Size::i32Bit, ARMEmitter::Reg::r1, TMP2);
```

Comment:
The JIT.cpp side already sign extends when writing into TMP2 so I don't think this is necessary

Reply:
You're right, frontend now doing the sign/zero-extend and the dispatcher only doing a move like the other ops.